runelite-client/src/main/java/net/runelite/client/plugins/groundmarkers/GroundMarkerSharingManager.java

```diff
@@ -69,6 +70,9 @@
 	private final ChatMessageManager chatMessageManager;
 	private final ChatboxPanelManager chatboxPanelManager;
 
+	private boolean menuOptionsAdded = false;
+	private boolean clearOptionAdded = false;
```

Comment:
I agree with your comment on these not being needed. I suspect we will not be changing the backing managed menu map in a way which would break this functionality. Hence, I'd simplify these operations to `menuManager.(add|remove)ManagedCustomMenu()`

Reply:
I've removed the booleans and the checks